plugins/modules/aci_endpoint_mac_tag.py

```diff
@@ -0,0 +1,339 @@
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str"),
+        name=dict(type="str"),
+        endpoint_mac_address=dict(type="str", aliases=["mac_addr", "mac"]),
+        bd=dict(type="str", aliases=["bd_name"]),
+        name_alias=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "bd", "endpoint_mac_address"]],
+            ["state", "present", ["tenant", "bd", "endpoint_mac_address"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    endpoint_mac_address = module.params.get("endpoint_mac_address")
+    bd = module.params.get("bd")
+    name = module.params.get("name")
+    name_alias = module.params.get("name_alias")
+    annotation = module.params.get("annotation")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    endpoint_mac_tag_rn = None
+    endpoint_mac_tag_module_object = None
+    if endpoint_mac_address and bd:
+        endpoint_mac_tag_rn = "eptags/epmactag-{0}-[{1}]".format(endpoint_mac_address, bd)
+        endpoint_mac_tag_module_object = "{0}-[{1}]".format(endpoint_mac_address, bd)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="fvEpMacTag",
+            aci_rn=endpoint_mac_tag_rn,
+            module_object=endpoint_mac_tag_module_object,
+            target_filter=dict(mac=endpoint_mac_address, bdName=bd),
```

Comment:
```suggestion
            target_filter=dict(mac=endpoint_mac_address, bdName=bd, name=name),
```
Should we add name here too?